Makefile

```diff
@@ -50,12 +53,34 @@ clean:
 	docker-compose stop
 	docker-compose down --remove-orphans --volumes --rmi all
 	docker-compose rm -svf
-	docker rmi --force openwisp/openwisp-base:latest \
-				openwisp/openwisp-base:intermedia-system \
-				openwisp/openwisp-base:intermedia-python \
-				`docker images -f "dangling=true" -q` || true
+	docker rmi --force `docker images -f "reference=*/openwisp-*" -q` \
+	                   `docker images -f "dangling=true" -q`
 
 # Publish
-publish: publish-build develop-runtests
-	docker push openwisp/openwisp-base:latest
+publish: develop-runtests
+	docker push $(REGISTRY)/openwisp-base:$(TAG)
 	docker-compose push
+	docker tag $(REGISTRY)/openwisp-base:$(TAG) $(REGISTRY)/openwisp-base:latest
+	docker push $(REGISTRY)/openwisp-base:latest
+	images=$$(docker-compose config | awk '{if($$0 ~ /image: $(REGISTRY)/ && !visited[$$0]++) print $$2}'); \
+	for image in $$images; do \
+		img_target=`echo $$image | sed "s/:$(TAG)/:latest/g"`; \
+		docker tag $$image $$img_target; \
+		docker push $$img_target; \
+	done
+
+travis-publish: TAG := nightly
+travis-publish: travis-runtests
+	echo "$(DOCKER_PASSWORD)" | docker login -u "$(DOCKER_USERNAME)" --password-stdin
```

Comment:
Just leaving a comment for myself : `docker rmi $(docker images |grep 'openwisp/')`